src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1050,10 +1053,27 @@ private void EmitCatchBlock(BoundCatchBlock catchBlock)
             }
 
             EmitBlock(catchBlock.Body);
-
             _builder.CloseLocalScope();
         }
 
+        private static bool IsFilterConstantFalse(BoundCatchBlock catchBlock)
+        {
+            // Depending on compilation options, BoundLiteral can be wrapped by BoundSequence or BoundSequencePointExpression
+            if(catchBlock.ExceptionFilterOpt != null)
+            {
+                if(catchBlock.ExceptionFilterOpt is BoundSequence sequence)
+                {
+                    return sequence.Value is BoundSequencePointExpression expression
```

Comment:
We should not mix two purposes of removing unreachable catch blocks

1) Fixing the bug by removal of formally unreachable catches (as per language spec and control flow rules). 
This is easier to do in lowering. - take a look at VisitCatchBlock in LocalRewriter_TryStatements.cs
Checking that the filter is a constant false and returning `null` should be sufficient to fix the bug.

2) Additional optimizations where the condition becomes `false` as a result of lowering and further optimizations. That would be harder to do and it would need to be done either in the codegen or in the optimizer pass. And only when /o+ flag is set.
This should be a separate item. Costs/Benefits/Risk for this part would not be as favorable as for the `#1` above.